GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -402,8 +403,19 @@ private bool TryGetSymLinkTarget(string sha, out string symLinkTarget)
                                 if (bufferIndex == buffer.Length)
                                 {
                                     bufferIndex = 0;
+                                    bufferSize += buffer.Length;
                                 }
                             }
+                            bufferSize += Convert.ToInt32(bufferIndex);
+
+                            if (bufferSize != blobLength)
+                            {
+                                // If the read size does not match the expected size print an error and add the file to ModifiedPaths.dat
+                                // This allows the user to see that something went wrong with file hydration
+                                // Unfortunitely we must do this check *after* the file is hydrated since the header isn't corrupt for trunctated objects on mac
+                                this.OnLogError($"Read {relativePath} to {bufferSize}, not expected size of {blobLength}");

```

Comment:
`this.OnLogError` is a callback from the ProjFS user-mode library.  

Rather than calling `this.OnLogError` ourselves we can use `this.Context.Tracer.RelatedError` (which keeps `OnLogError` specifically for logging errors from ProjFS)